src/map/skill.c

```diff
@@ -14052,11 +14052,51 @@ int skill_check_condition_castbegin(struct map_session_data* sd, uint16 skill_id
 			}
 			break;
 		case RA_WUGDASH:
-			if(!pc_isridingwug(sd)) {
-				clif->skill_fail(sd,skill_id,USESKILL_FAIL_CONDITION,0);
+			if (!pc_isridingwug(sd)) {
+				clif->skill_fail(sd, skill_id, USESKILL_FAIL_CONDITION, 0);
 				return 0;
 			}
+			else {
+				int16 sx = sd->bl.x, sy = sd->bl.y;
+				uint8 dir = (unit->getdir(&sd->bl)) % 8;
+
+				switch (dir) {
```

Comment:
You could use `dirx` and `diry` here instead (they're defined in unit.h, search skill.c, unit.c map.c, script.c for various examples of use). Something like:

```
int dir = unit->getdir(&sd->bl);
int16 sx = sd->bl.x + dirx[dir % 8];
int16 sy = sd->bl.y + diry[dir % 8];
```